content/altinn-api/clients/_index.md

```diff
@@ -0,0 +1,201 @@
+
+Lastly to be examined from the folder, is the error file on the failed instantiation (`error-01036800236.txt`) from our example:
+
+```txt
+Status code '400', error message: "Instance owner lookup failed."
+```
+
+*With purpose we tried to use a person number which does not exist in database to generate an error.*
+
+## Behind the scenes
+
+Now we will take a short look of what happens behind the scenes when running the example application.
+
+Our example client application iterates over the XML files in the folder, creates an instance template containing an instanceOwnerLookup.PersonNumber value to find, and then generate the multipart content with the following blocks of C# code.
+
+XML file iteration and instanceOwnerLookup creation:
+
+```c#
+foreach (string xmlFilePath in xmlFilePaths)
+            {
+                xmlFileName = xmlFilePath.Split("\\").Last();
+
+                // The person number is the XML filename
+                personNumber = xmlFileName.Split(".").First();
+
+                Instance instanceTemplate = new Instance()
+                {
+                    InstanceOwnerLookup = new InstanceOwnerLookup()
+                    {
+                        PersonNumber = personNumber,
+                    }
+                };
+```
+
+Multipart content creation:
+
+```c#
+MultipartFormDataContent content = new MultipartContentBuilder(instanceTemplate)
+                .AddDataElement("default", new FileStream(xmlFilePath, FileMode.Open), "application/xml")
+                .Build();
+```
+
+*We have made a helper class, `MultipartContentBuilder`, to easily add new data elements to the multipart in case several attachements are to be used for prefill.*
+
+In our example we have used an element type "`default`" and content type "`application/xml`". These two must be declared in the application metadata in order to store the files. The same goes for "`instance`" and "`application/json`" to create the instance, which is added within the constructor of `MultipartContentBuilder`:
+
+```c#
+public class MultipartContentBuilder
+    {
+        private MultipartFormDataContent builder;
+
+        public MultipartContentBuilder(Instance instanceTemplate)
+        {
+            builder = new MultipartFormDataContent();
+            if (instanceTemplate != null)
+            {
+                StringContent instanceContent = new StringContent(JsonConvert.SerializeObject(instanceTemplate), Encoding.UTF8, "application/json");
+
+                builder.Add(instanceContent, "instance");
+            }
+        }
+```
+
+The C# code then generates the following multipart information to be sent with the request to the Altinn API:
+
+```http
+Content-Type: multipart/form-data; boundary="abcdefg"
+
+--abcdefg
+Content-Type: application/json; charset=utf-8
+Content-Disposition: form-data; name="instance"
+
+{
+  "instanceOwnerLookup":  { "personNumber": "01036800298" },
+  "createdDateTime": "0001-01-01T00:00:00"
```

Comment:
Must be removed and storage interface updated to handle createdDateTime to nullable.

Reply:
Removed. Will make a new issue regarding nullable DateTime in Storage.Interface > Models > "Instance.cs".